components/TextFields/src/MDCTextField.m

```diff
@@ -106,6 +111,72 @@ - (void)commonMDCTextFieldInitialization {
                       object:self];
 }
 
+#pragma mark - Underline View Implementation
+
+- (void)setupUnderlineConstraints {
+  NSLayoutConstraint *underlineLeading = [NSLayoutConstraint constraintWithItem:self.underline
+                               attribute:NSLayoutAttributeLeading
+                               relatedBy:NSLayoutRelationEqual
+                                  toItem:self
+                               attribute:NSLayoutAttributeLeading
+                              multiplier:1
+                                                                       constant:0];
+  underlineLeading.priority = UILayoutPriorityDefaultLow;
+  underlineLeading.active = YES;
+
+  NSLayoutConstraint *underlineTrailing = [NSLayoutConstraint constraintWithItem:self.underline
+                               attribute:NSLayoutAttributeTrailing
+                               relatedBy:NSLayoutRelationEqual
+                                  toItem:self
+                               attribute:NSLayoutAttributeTrailing
+                              multiplier:1
+                                                                        constant:0];
+  underlineTrailing.priority = UILayoutPriorityDefaultLow;
+  underlineTrailing.active = YES;
+
+  CGFloat estimatedTextHeight = MDCCeil(self.font.lineHeight * 2.f) / 2.f;
```

Comment:
CJK support will be added later. Right now it's guaranteed on latin, arabic and hebrew.